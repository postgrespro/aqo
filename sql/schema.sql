DROP SCHEMA IF EXISTS test CASCADE;

-- Check Zero-schema path behaviour
CREATE SCHEMA IF NOT EXISTS test;
SET search_path TO test;
DROP SCHEMA IF EXISTS test CASCADE;
CREATE EXTENSION aqo;  -- fail

-- Check default schema switching after AQO initialization
CREATE SCHEMA IF NOT EXISTS test1;
SET search_path TO test1, public;
CREATE EXTENSION aqo;
SELECT true AS success FROM aqo_reset();
SET aqo.mode = 'intelligent';

CREATE TABLE test (id SERIAL, data TEXT);
INSERT INTO test (data) VALUES ('string');
SELECT * FROM test;

-- Check AQO service relations state after some manipulations
-- Exclude fields with hash values from the queries. Hash is depend on
-- nodefuncs code which is highly PostgreSQL version specific.
SELECT query_text FROM aqo_query_texts
ORDER BY (md5(query_text)) DESC;
SELECT learn_aqo, use_aqo, auto_tuning FROM aqo_queries
ORDER BY (learn_aqo, use_aqo, auto_tuning);
DROP SCHEMA IF EXISTS test1 CASCADE;
