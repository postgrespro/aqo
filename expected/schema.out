DROP SCHEMA IF EXISTS test CASCADE;
NOTICE:  schema "test" does not exist, skipping
-- Check Zero-schema path behaviour
CREATE SCHEMA IF NOT EXISTS test;
SET search_path TO test;
DROP SCHEMA IF EXISTS test CASCADE;
CREATE EXTENSION aqo;  -- fail
ERROR:  no schema has been selected to create in
-- Check default schema switching after AQO initialization
CREATE SCHEMA IF NOT EXISTS test1;
SET search_path TO test1, public;
CREATE EXTENSION aqo;
SELECT true AS success FROM aqo_reset();
 success 
---------
 t
(1 row)

SET aqo.mode = 'intelligent';
CREATE TABLE test (id SERIAL, data TEXT);
INSERT INTO test (data) VALUES ('string');
SELECT * FROM test;
 id |  data  
----+--------
  1 | string
(1 row)

-- Check AQO service relations state after some manipulations
-- Exclude fields with hash values from the queries. Hash is depend on
-- nodefuncs code which is highly PostgreSQL version specific.
SELECT query_text FROM aqo_query_texts
ORDER BY (md5(query_text)) DESC;
              query_text               
---------------------------------------
 COMMON feature space (do not delete!)
 SELECT * FROM test;
(2 rows)

SELECT learn_aqo, use_aqo, auto_tuning FROM aqo_queries
ORDER BY (learn_aqo, use_aqo, auto_tuning);
 learn_aqo | use_aqo | auto_tuning 
-----------+---------+-------------
 f         | f       | f
 t         | f       | t
(2 rows)

DROP SCHEMA IF EXISTS test1 CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to extension aqo
drop cascades to table test
